### PR TITLE
Update MistralAI Moonshot AI and Qwen parameters.

### LINKED ISSUE #8141: Please Updata MistralAI
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

add open-codestral-mamba  codestral-latest  open-mistral-nemo

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by AAEE86**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

Fixes #8141

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Review by crazywoola — CHANGES_REQUESTED** [review]
I don't think it supports 128k output tokens.
Output tokens `max tokens` is different from context tokens.

**Comment by AAEE86** [comment]
> I don't think it supports 128k output tokens. Output tokens is different from context tokens.`max tokens`

https://docs.mistral.ai/getting-started/models/
https://mistral.ai/news/mistral-large-2407/
Quicker_20240909_161833
QQ截图20240909162047
I referred to the materials provided officially.

**Comment by crazywoola** [comment]
> I don't think it supports 128k output tokens. Output tokens `max tokens` is different from context tokens.

Yes, but this is the context size from the docs, it did not mention the output tokens.

**Comment by AAEE86** [comment]
https://docs.mistral.ai/getting-started/models/ 
The MAX Tokens for this page should be the context size.

I didn't find max output tokens.

I think it can be...
32K context size Set the max token to 4096.
64K context size Set the max token to 8192.
128K context size Set the max token to 16384.
256K context size Set the max token to 32768.

**Comment by AAEE86** [comment]
> I don't think it supports 128k output tokens. Output tokens `max tokens` is different from context tokens.

The modification has been completed. Please review.

File: api/core/model_runtime/model_providers/moonshot/llm/moonshot-v1-8k.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ parameter_rules:
     use_template: max_tokens
     default: 512
     min: 1
-    max: 8192
+    max: 2048
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/moonshot/llm/moonshot-v1-8k.yaml]: > 这个值建议按需给个合理的值，如果不给的话，我们会给一个不错的整数比如 1024。特别要注意的是，这个 max_tokens 是指您期待我们返回的 token 长度，而不是输入 + 输出的总长度。比如对一个 moonshot-v1-8k 模型，它的最大输入 + 输出总长度是 8192，当输入 messages 总长度为 4096 的时候，您最多只能设置为 4096，否则我们服务会返回不合法的输入参数（ invalid_request_error ），并拒绝回答。如果您希望获得“输入的精确 token 数”，可以使用下面的“计算 Token” API 使用我们的计算器获得计数

https://platform.moonshot.cn/docs/api/chat#%E5%AD%97%E6%AE%B5%E8%AF%B4%E6%98%8E

**Pull request closed without merging**